web/source/osk/oskKey.ts

```diff
@@ -513,8 +500,19 @@ namespace com.keyman.osk {
     }
 
     public refreshLayout(vkbd: VisualKeyboard) {
-      if(this.label) { // space bar may not define the text span!
-        this.label.style.fontSize = this.getIdealFontSize(vkbd, this.btn.style);
+      // space bar may not define the text span!
+      if(this.label) {
+        if(!this.label.classList.contains('kmw-spacebar-caption')) {
+          this.label.style.fontSize = this.getIdealFontSize(vkbd, this.keyText, this.btn.style);
+        } else {
+          // Remove any custom setting placed on it before recomputing its inherited style info.
+          this.label.style.fontSize = '';
+          const fontSize = this.getIdealFontSize(vkbd, this.label.textContent, getComputedStyle(this.label), true);
+
+          // Since the kmw-spacebar-caption version uses !important, we must specify
+          // it directly on the element too; otherwise, scaling gets ignored.
+          this.label.style.setProperty("font-size", fontSize, "important");
```

Comment:
Is line 509 redundant given line 514? And if not, does that mean that line 506 will have problems if line 514 is ever run?

Reply:
See line 508 and contrast with this part of 510:  `getComputedStyle(this.label)`.

509 is preventing 510's "computed style" op from reusing a previous font size setting applied to the label.  This matters when it comes to resize events, rotation events, etc.  Key cap scaling should be based on the styles _inherited_ by the label element **_and from kmwosk.css_**.  This last part is the important thing to catch - `kmw-spacebar-caption` applies font-size scaling, while `kmw-key-text` does not:

https://github.com/keymanapp/keyman/blob/f3788d86be2ea6dcdc4420f964fb0653cf049764/web/source/resources/osk/kmwosk.css#L530

Well, re `kmw-key-text`, there is one exception:

https://github.com/keymanapp/keyman/blob/f3788d86be2ea6dcdc4420f964fb0653cf049764/web/source/resources/osk/kmwosk.css#L326-L328

So, when `kmw-key-text` _does_ set font-scaling, it's actually to _prevent_ scaling, so the base assumption of "1 em" unit default scaling holds everywhere... but for the spacebar caption.  Line 509 simply enforces this assumption for consistency.

Our existing key-text / caption scaling logic assumes that style information provided to it defines the original (and maximum) desired font size.  So... if the caption's scale was set in 'portrait' (which will generally require smaller size) and then set to landscape, that would likely cause landscape to reuse the smaller 'portrait' scaling rather than the scaling it would have had if the keyboard were first loaded in landscape.  That's an inconsistency that would appear buggy to end users.

Also, line 509 prevents us from having to combine 'old' and 'new' scaling with extra logic, like might occur going from landscape to portrait.  Setting the initial scale to 1 means we can let the browser handle those calculations for us.

------

Now, that said...

Reply:
Gotcha.